proof/refine/ARM/TcbAcc_R.thy

```diff
@@ -123,18 +123,34 @@ lemma valid_objs_valid_tcbE':
   apply (fastforce simp: projectKO_def projectKO_opt_tcb return_def valid_tcb'_def)
   done
 
```

Comment:
I'm generally in favour of introducing this predicate, and I think this implementation will be ok for now. An ideal implementation would make use of projections (depend on some `kheap_tcbs s` rather than `kheap s`).

Reply:
I'm slightly concerned about introducing some duplication of work because we already prove valid_objs' over everything. If we go ahead with this, we will need to keep an eye on that.